lifecycle_test.go

```diff
@@ -456,6 +456,83 @@ func TestMergePortBindings(t *testing.T) {
 	}
 }
 
+func TestPortMappingCheck(t *testing.T) {
+	makePortMap := func(ports []string) nat.PortMap {
+		out := make(nat.PortMap)
+		for _, port := range ports {
+			// We don't care about the actual binding in this test
+			out[nat.Port(port)] = make([]nat.PortBinding, 0)
+		}
+		return out
+	}
+
+	tests := []struct {
+		name                  string
+		exposedAndMappedPorts nat.PortMap
+		exposedPorts          []string
+		expectError           bool
+	}{
+		{
+			name:                  "No protocol given",
+			exposedAndMappedPorts: makePortMap([]string{"1024/tcp"}),
+			exposedPorts:          []string{"1024"},
+			expectError:           false,
+		},
+		{
+			name:                  "Protocol given",
+			exposedAndMappedPorts: makePortMap([]string{"1024/tcp"}),
+			exposedPorts:          []string{"1024/tcp"},
+			expectError:           false,
+		},
+		{
+			name:                  "Protocol and target port given",
+			exposedAndMappedPorts: makePortMap([]string{"1024/tcp"}),
+			exposedPorts:          []string{"1024:1024/tcp"},
+			expectError:           false,
+		},
+		{
+			name:                  "Only target port given, no protocol",
+			exposedAndMappedPorts: makePortMap([]string{"1024/tcp"}),
+			exposedPorts:          []string{"1024:1024"},
+			expectError:           false,
+		},
+		{
+			name:                  "Multiple ports with different variations",
+			exposedAndMappedPorts: makePortMap([]string{"1024/tcp", "1025/tcp", "1026/tcp"}),
+			exposedPorts:          []string{"1024", "25:1025/tcp", "1026:1026"},
+			expectError:           false,
+		},
+		{
+			name:                  "No ports mapped yet",
+			exposedAndMappedPorts: makePortMap([]string{}),
+			exposedPorts:          []string{"1024"},
+			expectError:           true,
+		},
+		{
+			name:                  "The wrong port has been mapped",
+			exposedAndMappedPorts: makePortMap([]string{"1023/tcp"}),
+			exposedPorts:          []string{"1024"},
+			expectError:           true,
+		},
+		{
+			name:                  "A subset of ports have been mapped",
+			exposedAndMappedPorts: makePortMap([]string{"1024/tcp", "1025/tcp"}),
+			exposedPorts:          []string{"1024", "1025", "1026"},
+			expectError:           true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := checkPortsMapped(tt.exposedAndMappedPorts, tt.exposedPorts)
+			if tt.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
```

Comment:
nit: I would use golden path flow as its easier to follow.

Either way I would recommend `require` instead of `assert`, no difference in this case but for more complex flows it eliminates knock on failures which just add noise, so good to standardise on.
```suggestion
			if tt.expectError {
				require.Error(t, err)
				return
			}
			
			require.NoError(t, err)
```

Reply:
https://github.com/testcontainers/testcontainers-go/pull/2658/commits/5fe2f77b1c6101b2c550f182ec4021d828ff5bd0